amacc.c

```diff
@@ -856,137 +1136,7 @@ void stmt()
-    case Sub:  gen((int *) n[1]); *++e = PSH; gen(n + 2); *++e = SUB; break;
-    case Mul:  gen((int *) n[1]); *++e = PSH; gen(n + 2); *++e = MUL; break;
-    case Syscall:
-    case Func:
-        c = b = (int *) n[1]; k = 0; l = 1;
-        // how many parameters
-        while (b && l) { ++k; if (!(int *) *b) l = 0; else b = (int *) *b; }
-        j = 0; a = malloc(sizeof(int *) * k); b = c; l = 1;
-        while (b && l) { a[j] = (int) b;  if (!(int *) *b) l = 0; else b = (int *) *b; ++j; }
-        if (j > 0) --j;
-        // push parameters
-        while (j >= 0 && k > 0) { gen(b + 1);  *++e = PSH; --j; b = (int *) a[j]; }
-        free(a);
-        if (i == Func) *++e = JSR; *++e = n[2];
-        if (n[3]) { *++e = ADJ; *++e = n[3]; }
-        break;
-    case While:
-        d = (e + 1);
-        gen((int *) n[1]); // condition
-        *++e = BZ; b = ++e;
-        gen(n + 2); // expression
-        *++e = JMP; *++e = (int) d;
-        *b = (int)(e + 1);
-        break;
-    case For:
-        gen((int *) n[4]);
-        a = (e + 1);
-        gen((int *) n[1]);
-        *++e = BZ; d = ++e;
-        *++e = JMP; c = ++e;
-        b = (e + 1);
-        gen((int *) n[2]); // area b
-        *++e = JMP; *++e = (int) a;
-        *c = (int)(e + 1);
-        gen((int *) n[3]); // area c
-        *++e = JMP; *++e = (int) b;
-        *d = (int)(e + 1);
-        break;
-    case Switch:
-        gen((int *) n[1]); // condition
-        a = cas; *++e = JMP; cas = ++e;
-        b = brks; d = def; brks = def = 0;
-        gen((int *) n[2]); // case statment
-        // deal with no default inside switch case
-        *cas = def ? (int) def : (int) (e + 1); cas = a;
-        while (brks) { a = (int *) * brks; *brks = (int)(e + 1); brks = a; }
-        brks = b; def = d;
-        break;
-    case Case:
-        *++e = JMP; ++e;
-        a = 0;
-        *e = (int) (e + 7); *++e = PSH; i = *cas; *cas = (int) e;
-        gen((int *) n[1]); // condition
-        if (e[-1] != IMM) fatal("bad case immediate");
-        *++e = SUB; *++e = BNZ; cas = ++e; *e = i + e[-3];
-        if (*(int *) n[2] == Switch) a = cas;
-        gen((int *) n[2]); // expression
-        if (a != 0) cas = a;
-        break;
-    case Break:
-        // set jump locate
-        *++e = JMP; *++e = (int) brks; brks = e;
-        break;
-    case Default:
-        def = e + 1;
-        gen((int *) n[1]); break;
-    case Return:  
-        if (n[1]) gen((int *) n[1]); *++e = LEV; break; // parse return ast 
-    case '{':  
-        gen((int *) n[1]); gen(n + 2); break;  // parse expression or statment from ast
-    case Enter:  *++e = ENT; *++e = n[1]; gen(n + 2); if (*e != LEV) *++e = LEV; break;
-    default:
-        if (i != ';') { printf("%d: compiler error gen=%d\n", line, i); exit(-1); }
-    }
-}
-
-void die(char *msg) { printf("%s\n", msg); exit(-1); }
+void die(char *msg) { printf("codegen: %s\n", msg); exit(2); }
```

Comment:
Why was it referred to `codegen`?

Reply:
Oh! This is my mistake since git rebase command.